consensus/polybft/state.go

```diff
@@ -59,6 +59,8 @@ const (
 	stateSyncMainBundleSize = 10
 	// number of stateSyncEvents to be grouped into one StateTransaction
 	stateSyncBundleSize = 1
+	// latest proposer calculator snapshot key
+	proposerCalcSnapshotKey = 7
```

Comment:
If I'm correct, you are basically storing the latest snapshot? Couldn't you use some of the attributes from the `ProposerSnapshot` (like `Height` or `Height|Round`) itself instead of relying on this constant?

Also you mentioned that you want this storage to be double buffered (preserving two instances), which is not the case currently I'd say. 
If you would rely on some unique attribute value instead, you could achieve that.